pkg/kubelet/gpu/nvidia/nvidia_gpu_manager.go

```diff
@@ -97,8 +98,8 @@ func (ngm *nvidiaGPUManager) Start() error {
 	if !os.IsNotExist(err) {
 		ngm.defaultDevices = append(ngm.defaultDevices, nvidiaUVMToolsDevice)
 	}
-
-	if err := ngm.discoverGPUs(); err != nil {
+	
+	if err := ngm.discoverGPUs("/proc/bus/pci/devices", "/dev"); err != nil {
```

Comment:
Let's not fail Start() if gpu discovery fails initially. Instead log discovery errors and try again. You can start a go thread that will continue to discover GPUs until it succeeds.

```go
go func() {
  var err error
  for {
    if err := ngm.discoverGPUs(...); err == nil {
       return
   }
   glog.Errorf("failed to discover Nvidia GPUs: %v", err)
   time.Sleep(time.Minute)
 }
}()
```